sql/parser/datum.go

```diff
@@ -237,7 +237,28 @@ func (d DFloat) IsMin() bool {
 }
 
 func (d DFloat) String() string {
-	return strconv.FormatFloat(float64(d), 'g', -1, 64)
+	s := strconv.FormatFloat(float64(d), 'g', -1, 64)
+	if parsesAsInt(s) {
```

Comment:
That link doesn't work (you need to get a shareable link). I think you could adjust the precision based on whether there is a non-zero fractional component:

```
  prec := -1
  if _, frac := math.Modf(float64(d)); frac == 0 {
    prec = 1  // Force 1 digit after the decimal
  }
  return strconv.FormatFloat(float64(d), 'g', prec, 64)
```